supabase/functions/auth/index.ts

```diff
@@ -0,0 +1,151 @@
+import { serve } from "https://deno.land/std/http/server.ts";
+import { createClient } from "https://esm.sh/@supabase/supabase-js";
+import { db } from "../_shared/db.ts";
+import { students } from "../_shared/schema.ts";
+import { getCookies, setCookie } from "https://deno.land/std/http/cookie.ts";
+
+type ValidationSchema = {
+  [key: string]: {
+    required?: boolean;
+    min?: number;
+    enum?: string[];
+  };
+};
+
+//need to move out the client create
+const supabase = createClient(
```

Comment:
Move `createClient` to `/_shared` directory as it will be re-used across multiple functions.